Add new blog post

davidmirror-ops opened the pull request:
Signed-off-by: David Espejo <daespejo@vmware.com>


**What this PR does / why we need it**:
Add august post to the blog

**Which issue(s) this PR fixes**
N/A
- Fixes #

**Special notes for your reviewer**:

**Release note**:
```
release-note
```

GuessWhoSamFoo commented on site/content/posts/2021-08-12-octant-0.23-release.md:
```diff
@@ -0,0 +1,91 @@
+
+While there are five possible values for Pod phase:
+
+| Value | Description |
```
Tables are rendering like so:

![image](https://user-images.githubusercontent.com/10288252/132587120-0ee20351-e813-4a93-bf9d-6eea090a7b96.png)

GuessWhoSamFoo commented on site/content/posts/2021-08-12-octant-0.23-release.md:
```diff
@@ -0,0 +1,91 @@
+excerpt: With release 0.23, Octant is a more accurate tool for troubleshooting scenarios
+author: David Espejo
+author_avatar: /img/contributors/david-espejo.jpg
+date: 2021-08-31
```
Filename and date mismatch

GuessWhoSamFoo commented on site/content/posts/2021-08-12-octant-0.23-release.md:
```diff
@@ -0,0 +1,91 @@
+
+# POD lifecycle details
+
+According to the [oficial docs](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/), the Kubernetes API has a PodStatus object with a field called **Phase** but there's also the container **State** field that Kubernetes tracks. The difference between the two lies in the level the API server is reffering to: the Pod or the container respectively.
```
spelling: official

spelling: referring

GuessWhoSamFoo commented on site/content/posts/2021-08-12-octant-0.23-release.md:
```diff
@@ -0,0 +1,91 @@
+
+![](/img/posts/2021/08/18/octantStatus01.png)
+
+We can see that it's in the **Running** phase but the actual container state is **CrashLoopBackOff** which means that the Kubelet has succesfully started it several times but it has crashed afterwards without reaching a stable state yet.
```
spelling: successfully

GuessWhoSamFoo commented on site/content/posts/2021-08-12-octant-0.23-release.md:
````diff
@@ -0,0 +1,91 @@
+```json
+    {"severity":"ERROR","timestamp":"2021-08-31T21:08:56.7648856Z","logger":"activator","caller":"websocket/connection.go:145","message":"Websocket connection could not be established","commit":"c75484e","knative.dev/controller":"activator","knative.dev/pod":"activator-67656dcbbb-hq62r","error":"dial tcp 10.96.38.80:8080: connect: connection refused"}
+```
+There it is what seems to be root cause of the issue: a network connectivity problem. 
````
That's not a root cause. Why is the connection refused?

davidmirror-ops replied:
Still working on this

GuessWhoSamFoo commented on site/content/posts/2021-08-12-octant-0.23-release.md:
```diff
@@ -0,0 +1,91 @@
+
+# Octant and the Status field
+
+Octant has always displayed the **Phase** field for each resource, but with the release of version 0.23, the team added the **Status** field which -in the case of Pods- refers specifically to the container State. 
```
We want to clarify that `Status` shown in kubectl takes into account the container state which is why it differs from `Phase`.

davidmirror-ops replied:
Still working on better phrasing for this

xtreme-vikram-yadav commented on site/content/posts/2021-08-12-octant-0.23-release.md:
```diff
@@ -0,0 +1,91 @@
+|Running| Container executing with no issues nor pending tasks|
+|Terminated| Container either ran to completion or it failed
+
+So, coming back to our example: what kubectl is showing for the "activator" Pod is the **Phase** field (in this case **Running** as the container has been created) but the application there it's not usable yet. How could we determine more accurately the actual situation of the Pod?
```
Minor update: `but the application there it's not usable yet` -> `but the application there is not usable yet`

davidmirror-ops commented:
I just hope I didn't screw up again my branch and everything may look better for this to be merged.

GuessWhoSamFoo commented:
The blog post was added twice as two different file names.

Tangential side note, the preview image for one of the resources is also down 
![image](https://user-images.githubusercontent.com/10288252/133527996-5296e647-6d59-41fc-aabe-957a1709c945.png)

davidmirror-ops commented:
Fixed a missing commit to actually delete the old blog post file, also added the missing image. Not exactly an atomic PR but locally tested and hoping it's correct

GuessWhoSamFoo commented on site/content/posts/2021-08-31-octant-0.23-release.md:
```diff
@@ -0,0 +1,112 @@
+| Failed | All containers terminated and at least one terminated in failure|
+| Unknown| Most likely the node is offline |
+
+there are also three **container states** defined in the Kubernetes API:
```
Nit: Capitalize

GuessWhoSamFoo commented on site/content/posts/2021-08-31-octant-0.23-release.md:
````diff
@@ -0,0 +1,112 @@
+
+![](/img/posts/2021/08/18/octantStatus06.png)
+
+5. In this case, the memory request was modified to meet the same value (```100Mi```) all the additional Pods in the app have requested and, in consequence, ```autoscaler``` was able to run and the ```activator``` Pod finally reached the **Running**  state:
````
Is this saying setting a request equal to a limit will cause a crash loop or was the image intended to how 100Mi?

davidmirror-ops commented:
New commit to address review feedback and new results from root cause analysis
